### PR TITLE
AddRedisCache extension method to IServiceCollection 

### PR DESCRIPTION
**Pull request by unaizorrilla**
AddRedisCache extension method in order to enable Redis service.

### CODE REVIEW

File: test/Microsoft.Framework.Caching.Redis.Tests/CacheServiceExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.DependencyInjection;
+using System;
```
Comment (Tratcher) [inline test/Microsoft.Framework.Caching.Redis.Tests/CacheServiceExtensionsTests.cs]: System usings should go first. There's a VS option for that.

File: test/Microsoft.Framework.Caching.Redis.Tests/CacheServiceExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Framework.Caching.Redis
+{
+    public class CacheServiceExtensionsTests
+    {
+
```
Comment (Tratcher) [inline test/Microsoft.Framework.Caching.Redis.Tests/CacheServiceExtensionsTests.cs]: nit: whitespace

**Comment by Tratcher** [comment]
Looks good.

File: src/Microsoft.Framework.Caching.Redis/CachingServiceExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+using Microsoft.Framework.Caching.Redis;
+
+namespace Microsoft.Framework.DependencyInjection
+{
+    public static class CachingServicesExtensions
+    {
+        public static IServiceCollection AddRedisCache(this IServiceCollection collection)
+        {
+            collection.AddOptions();
+            collection.TryAdd(ServiceDescriptor.Transient<IDistributedCache, RedisCache>());
```
Comment (kichalla) [inline src/Microsoft.Framework.Caching.Redis/CachingServiceExtensions.cs]: curious if this should be a Singleton?...the current Redis cache implementation, which depends on StackExchange Redis client, talks about caching the ConnectionMultiplexer once its created...was just reading this : https://github.com/StackExchange/StackExchange.Redis/blob/master/Docs/Basics.md

Comment (Tratcher) [inline src/Microsoft.Framework.Caching.Redis/CachingServiceExtensions.cs]: We do the same thing here: https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Framework.Caching.Distributed/CachingServicesExtensions.cs#L14
This needs to be worked out across all of caching. It's not a problem for this PR.

Comment (kichalla) [inline src/Microsoft.Framework.Caching.Redis/CachingServiceExtensions.cs]: Sure :+1:

**Comment by unaizorrilla** [comment]
Updated code directly on github

File: src/Microsoft.Framework.Caching.Redis/CachingServiceExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.Caching.Redis;
+namespace Microsoft.Framework.DependencyInjection
```
Comment (Eilon) [inline src/Microsoft.Framework.Caching.Redis/CachingServiceExtensions.cs]: Code style: can you add an empty line before the namespace? (In other files too)

**Comment by Eilon** [comment]
@unaizorrilla can you make the small style change I suggested and then squash your commits? That'll make it easier for us to merge and have a clean history.

**Comment by unaizorrilla** [comment]
Ok, Elion, If you prefer I can do a new pr instead of squash

**Comment by Eilon** [comment]
@unaizorrilla no need for a new PR. Just rebase (if needed) and [squash](http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) the commits.

**Comment by jasonwilliams200OK** [comment]
@unaizorrilla,

```powershell
# powershell with git

# after making the changes

cd into/caching/dir
git add .    # adds your latest changes, for example, addressing Elion's comment
git commit -am "comment fixes (temp message)"  # pseudo random commit message,
                                               # which will soon be squashed
git remote add aspnet-caching https://github.com/aspnet/Caching/  # add remote to upstream
git pull --rebase aspnet-caching  # rebase your work on top of the history
git rebase -i HEAD~6  # by now total number of commits in this PR are 6
#
# you have entered git interactive rebase mode
# editor like notepad is opened. You are offered 6 commits
# to choose from.
# Except for the first line, replace pick with `f` (fixup)
# for the next 5 lines
# if you want to change the commit log for the final single commit,
# replace pick with `r` (reword) for the first line
# save and close editor (such as notepad)
#
# if you had chosen `r` for the first line, at this point you will get another notepad
# where you can change the commit log.
# There is one standard pretty way of writing a commit log:
# 
# <feaure-name>: commit message (all within 50 chars, the rest of description goes in next line)
# for example:
#
# redis: Adds extension method AddRedisCache.
# This commit adds an extension method called
# `AddRedisCache` in order to enable Redis service.
#
# Issue-URL: <if-any>.
# PR-URL: #65.
# Reviewed-By: @Eillon, @Trancher.
#
#
# after that save and close notepad and in shell:

git push -f   # rewrites the history
```

Note 1: we cannot commit message the first editor opens for configuring commit. `reword` or `r` option should be set. If there are multiple rewords, it will separately open the editor for each subsequently after you close the first one.

Note 2: The commit log message format is the most approved format in many repositories on GitHub, including [joyent/node](https://github.com/joyent/node), [libuv/libuv](https://github.com/libuv/libuv) etc. (I really wish if ASP team will inherit this convention and apply it across the board: https://github.com/libuv/libuv/blob/v1.x/CONTRIBUTING.md#commit).

Now in the commit history on GitHub, you will find your commit with brief part of description visible and the rest of the description is the following line (after expanding the ellipsis <kbd>...</kbd>).

**Comment by davidfowl** [comment]
@Tratcher What's the deal with this PR? It's out of date so if we're taking it we need a rebase

**Comment by jasonwilliams200OK** [comment]
Too bad to see that for minor nits, the PRs under aspnet/* are having merge conflicts and getting staled unmerged for months! In some repos, like libuv/libuv and now on dotnet/corefx as well, the repo owners commit on behalf of contributor for minor things rolling on with the flow.

Please see this related conversation: https://github.com/dotnet/corefx/pull/1721#issuecomment-109714893.

**Comment by davidfowl** [comment]
@jasonwilliams200OK yah we kinda suck. This change is easy enough to rebase on our own (potentially). We've been letting PRs sit for too long in the midst of large code churn making them harder to merge when we get do them. That's all going to change though. 

Generally, though, we want the authors of the PR to handle merge conflicts as they would best understand how to resolve the conflicts.

**Comment by Tratcher** [comment]
@kichalla Can you rebase, squash, validate, and merge this?

**Comment by davidfowl** [comment]
@Tratcher it says cla-signed but I don't see CLA approved..

**Comment by kichalla** [comment]
a8790019715bca25555238ef4a792960d3ca1e4c

**Pull request closed without merging**